rdr_service/dao/study_nph_dao.py

```diff
@@ -474,6 +500,63 @@ def __init__(self):
     def get_id(self, obj: OrderedSample):
         return obj.id
 
+    @staticmethod
+    def _is_order_cancelled(order: Order) -> bool:
+        return order.status == "cancelled"
+
+    @staticmethod
+    def _is_ordered_sample_cancelled(ordered_sample: OrderedSample) -> bool:
+        return ordered_sample.status == "cancelled"
+
+    def get_biospecimens_for_order(self, nph_participant: Participant, order: Order) -> Iterator[Dict[str, Any]]:
+        nph_stored_sample_session = NphStoredSampleDao()
+        nph_study_category_dao = NphStudyCategoryDao()
+        with self.session() as session:
+            ordered_samples_for_participant: Iterable[OrderedSample] = list(
+                session.query(OrderedSample).filter(OrderedSample.order_id == order.id).all()
+            )
+            for ordered_sample in ordered_samples_for_participant:
+                parent_study_category = nph_study_category_dao.get_parent_study_category(order.category_id)
+                nph_module_id = nph_study_category_dao.get_parent_study_category(parent_study_category.id)
+                sample_processing_ts = ordered_sample.collected if not ordered_sample.parent is None else None
+                collectionDateUTC = _format_timestamp((ordered_sample.parent or ordered_sample).collected)
+                processingDateUTC = _format_timestamp(sample_processing_ts)
+                finalizedDateUTC = _format_timestamp(ordered_sample.finalized) if ordered_sample.finalized else None
+                sample_is_cancelled = (
+                    self._is_order_cancelled(order) or
+                    self._is_ordered_sample_cancelled(ordered_sample)
+                )
+                kit_id = ""
+                if (ordered_sample.identifier or ordered_sample.test).startswith("ST"):
+                    kit_id = order.nph_order_id
+
+                sample_status = "Cancelled" if sample_is_cancelled else "Active"
+                biospecimen_dict = {
+                    "orderID": order.nph_order_id,
+                    "visitID": parent_study_category.name if parent_study_category else "",
```

Comment:
Should this be `None` as well?

Reply:
`visitID` was set as empty string in the nph biobank export job, if the `parent_study_category` is None. Hence, I had to replicate the same.